lib/connection_pool/timed_stack.rb

```diff
@@ -111,6 +121,30 @@ def length
 
   private
 
+  def reap_connections!
+    @mutex.synchronize do
+      required_last_access = Time.now - @reap_after
```

Comment:
Maybe it'd be a better idea to reuse what's already available?

```suggestion
      required_last_access = current_time - @reap_after
```